src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,47 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  static alive = [];
+
+  constructor(name, health = 100) {
+    this.health = health;
+    this.name = name;
+  }
+
+  static addAnimal(animal) {
+    Animal.alive.push(animal);
+  }
+
+  static removeAnimal() {
+    Animal.alive = Animal.alive.filter((animal) => animal.health > 0);
+  }
 }
 
 class Herbivore extends Animal {
-  // write your code here
+  constructor(name, health = 100, hidden = false) {
+    super(name, health);
+    this.hidden = hidden;
+    Animal.addAnimal(this);
+  }
+
+  hide() {
+    this.hidden = true;
+  }
 }
 
 class Carnivore extends Animal {
-  // write your code here
+  constructor(name, health = 100) {
+    super(name, health);
+    Animal.addAnimal(this);
+  }
+
+  bite(animal) {
+    if (animal instanceof Herbivore && !animal.hidden) {
```

Comment:
The `bite` method will not work correctly if it's called with another carnivore. Add a check to ensure that the `animal` parameter is not an instance of `Carnivore`.